### DISCUSSION TITLE
Additional parameters work different compared to v8

### DISCUSSION THREAD

**Discussion by baennjer** [benzino77/tasmocompiler#287] [General]
When I enter in v9.0.1 of TasmoCompiler under item 4 (additional parameters) i.e. 
#define USE_SHUTTER
#define USE_ETHERNET
the result has the status "failed" error code 1

The input needs the following syntax to have a successfull compiled .bin file:
#ifdef USE_SHUTTER
 #undef USE_SHUTTER
#endif
#define USE_SHUTTER
#ifdef USE_ETHERNET
 #undef USE_ETHERNET
#endif
#define USE_ETHERNET

This behaviour is only to see when I compile v12.0.x of Tasmota. With v11.x.x all works fine with the "old" syntax...

**Comment by benzino77** [comment]
Please have a look at those places (especially examples provided there):
[Tasmota documentation](https://tasmota.github.io/docs/Compile-your-build/#changing-default-settings)
[TasmoCompiler instructions](https://github.com/benzino77/tasmocompiler#how-to-use-custom-parameters)

There is also a helping "placeholder" in "Custom parameters" step to help user properly define custom settings:
![image](https://user-images.githubusercontent.com/15658235/174955387-27f5fa81-800d-455f-a591-9a32789dc282.png)

**Reply by baennjer** [reply]
Cool docu, never seen it before. I was brought to TasmoCompiler from a random blog web site and there was just the short form described (i.e. #define USE_ETHERNET).
Thus I thought in v9.x.x short form is still working. Afterwards I noticed the placeholder as can be seen in your screenshot above.
Thanks for clarification!!!!

**Reply by benzino77** [reply]
No problem! Happy Tasmotizing ;)
